placeholder_accessors/index.md

```diff
@@ -192,6 +193,51 @@ The handler gains a new method,
 that registers the requirements of the placeholder accessor on the given
 command group.
 
+### Placeholder `accessor` without a buffer
+
+Not providing either the handler or the memory object
+makes the accessor default constructible,
+creating a "null" accessor.
+A null accessor can be bound to a buffer
+by constructing a new placeholder accessor with the buffer
+and assigning it to the null accessor.
+Example:
+
+```cpp
+accessor<T, dim, mode, target, access::placeholer::true_t> acc0;
+
+acc0 = accessor<T, dim, mode, target, access::placeholer::true_t>(buf);
+```
+
+To simplify placeholder accessor construction,
+a new free function `cl::sycl::codeplay::make_placeholder_accessor` is provided,
+which takes a buffer as an argument:
+
+```cpp
+template <access::mode mode, access::target target, typename T, int dim,
+          typename AllocatorT>
+accessor<T, dim, mode, target, access::placeholder::true_t>
+make_placeholder_accessor(buffer<T, dim, AllocatorT>& bufferRef);
+```
+
+A null accessor does not need to be registered with the command group,
+i.e. there is no need to call `require()` on it.
+It is allowed to pass a null accessor to a kernel,
+as long as it's not dereferenced inside the kernel,
+otherwise an out-of-bounds access will occur,
+likely leading to a segmentation fault or similar.
+If the null accessor has been bound to a buffer,
+it is no longer a null accessor,
+which means it has to be registered with the command group
+and can be dereferenced inside the kernel.
+
+An accessor can be checked for the existence of an associated buffer
+by calling the member function `is_null`.
+
+|Member function      |Description                                              |
+|---------------------|---------------------------------------------------------|
+|bool is_null() const |Returns false if the accessor is associated with a buffer, true otherwise.|
```

Comment:
What is the expected output of is_null in the following cases?

Case 1:
```cpp
accessor<T, dim, mode, target, access::placeholer::true_t> acc0;

q.submit([&](handler& h) {
      h.require(acc0, bufA);
});

// What is the output here?
std::cout << acc.is_null() << std::endl;
```

Case 2:
```cpp
accessor<T, dim, mode, target, access::placeholer::true_t> acc0;

std::thread t([](){
  q.submit([&](handler& h) {
      h.require(acc0, bufA);
  });
});
// What is the output here?
std::cout << acc.is_null() << std::endl;
t.join();
```

Reply:
This PR doesn't have that overload of the `require` function. The existing function doesn't modify the state of the accessor, and having an overload that would modify it is a dangerous inconsistency. That also means that both your cases would print true.
```cpp
h.require(acc0, bufA);
// is instead
acc0 = accessor<T, dim, mode, target, access::placeholer::true_t>(bufA);
h.require(acc0);
// or
acc0 = make_placeholder_accessor<mode, target>(bufA);
h.require(acc0);
```